internal/deploy/parameters_test.go

```diff
@@ -48,3 +55,258 @@ func TestChangeSet_getStepWeight(t *testing.T) {
 		})
 	}
 }
+
+func TestChangeSet_getVolumeName(t *testing.T) {
+
+	tests := []struct {
+		name    string
+		set     ChangeSet
+		action  func(set *ChangeSet)
+		want    string
+		wantErr string
+	}{
+		{
+			name: "valid volume name",
+			set:  ChangeSet{volume: stringRef("pvc-1")},
+			want: "pvc-1",
+		},
+		{
+			name:    "invalid volume name",
+			set:     ChangeSet{},
+			wantErr: `"volume" missing`,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			volumeName, err := tt.set.getVolumeName()
+			if len(tt.wantErr) > 0 {
+				require.NotNil(t, err)
+				require.Equal(t, tt.wantErr, err.Error())
+				return
+			}
+			require.Nil(t, err)
+			require.Equal(t, tt.want, volumeName)
+		})
+	}
+}
+
+func TestChangeSet_getVolumes(t *testing.T) {
+
+	tests := []struct {
+		name    string
+		set     ChangeSet
+		action  func(set *ChangeSet)
+		want    []v1.Volume
+		wantErr string
+	}{
+		{
+			name:    "volume name not included",
+			set:     ChangeSet{},
+			wantErr: `"volume" missing`,
+		},
+		{
+			name: "no volume source",
```

Comment:
Do tests `no volume source`and `incorrect volume source` test the same thing as `valid volume`?

Reply:
`valid volume` will check specifically for the existence of VolumeSource of type PersistentVolumeClaim while the other two error tests either lack PersistentVolumeClaim (its a ConfigMap instead) or where PersistentVolumeClaim does not exist